lib/containers/utils.pm

```diff
@@ -179,4 +179,23 @@ sub test_built_img {
     assert_script_run("rm -rf /root/templates");
 }
 
+=head2 can_build_sle_base
+
+C<can_build_sle_base> should be used to identify if sle base image runs against a
+system that it does not support registration and SUSEConnect B<for job running on OSD>.
+In this case the build of the container engine is not going to work.
+
+The call should return false if you try to test sle base container on osd but host is not sle
+true otherwise.
+
+=cut
+sub can_build_sle_base {
+    my $img         = shift;
+    my $is_sle_base = (ref $img eq 'ARRAY' ? $img->[0] : $img) =~ /^registry.suse.de/;
```

Comment:
What is actually needed here is something like `has_sle_registration`. The caller already knows whether the image to test needs that or not.

Reply:
i hope i get that right.